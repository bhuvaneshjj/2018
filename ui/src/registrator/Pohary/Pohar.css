.Pohar {
  margin-left: 5px;
}
