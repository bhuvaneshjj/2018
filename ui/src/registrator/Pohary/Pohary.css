.Pohary__div {
  padding-left: 30px;
  padding-right: 50px;
}

.Pohary__buttons {
  margin-left: 1em;
}

.Pohary__button {
  margin-left: 5px;
}
