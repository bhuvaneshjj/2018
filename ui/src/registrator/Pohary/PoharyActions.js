import { API_POHAR_PREDAN } from '../../common';
import { ActionPrefixes } from '../../constants';
import { WS_API } from '../../store/wsAPI';

export const POHAR_NAROK = 'nárok';
export const POHAR_NEPREVZATO = 'nepřevzato';
export const POHAR_PREDANO = 'předáno';
export const TYPY_POHARU = [POHAR_NAROK, POHAR_NEPREVZATO, POHAR_PREDANO];

const actionPrefix = ActionPrefixes.POHARY;
export const narokovanePrihlaskouFilterChange = () => ({
  type: `${actionPrefix}_NAROKOVANE_PRIHLASKOU_FILTER_CHANGE`
});
export const narokovaneStartemFilterChange = () => ({
  type: `${actionPrefix}_NAROKOVANE_STARTEM_FILTER_CHANGE`
});
export const neprevzateFilterChange = () => ({ type: `${actionPrefix}_NEPREVZATE_FILTER_CHANGE` });

// :TODO: API podporuje jen nepřevzato -> předáno. Je třeba dodělat i předáno -> nepřevzato.
export const isTypeDraggable = type => type === POHAR_NEPREVZATO;
export const isTypeDroppable = type => type === POHAR_PREDANO;

export const canDrop = ({ source, destination }) => {
  if (!isTypeDraggable(source.type)) {
    return false;
  }
  if (!isTypeDroppable(destination.type)) {
    return false;
  }
  if (source.id !== destination.id) {
    return false;
  }
  return source.type !== destination.type;
};

export const POHAR_PREDAN = 'POHAR_PREDAN';
export const poharPredan = ({ id }) => ({
  [WS_API]: {
    type: POHAR_PREDAN,
    endpoint: API_POHAR_PREDAN,
    request: { id },
    title: 'předávání poháru'
  }
});

export const onDrop = ({ source, destination, dispatch }) => {
  if (source.type === POHAR_NEPREVZATO && destination.type === POHAR_PREDANO) {
    dispatch(poharPredan({ id: source.id }));
    return undefined;
  }

  throw new Error('Nečekaný pohár.');
};

export const createOnDrop = dispatch => ({ source, destination }) =>
  onDrop({ source, destination, dispatch });
