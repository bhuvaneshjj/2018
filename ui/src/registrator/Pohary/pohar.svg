<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!-- Created with Inkscape (http://www.inkscape.org/) -->

<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   width="136.8098mm"
   height="167.65932mm"
   viewBox="0 0 136.8098 167.65932"
   version="1.1"
   id="svg8"
   inkscape:version="0.92.3 (2405546, 2018-03-11)"
   sodipodi:docname="pohar.svg">
  <defs
     id="defs2" />
  <sodipodi:namedview
     id="base"
     pagecolor="#ffffff"
     bordercolor="#666666"
     borderopacity="1.0"
     inkscape:pageopacity="0.0"
     inkscape:pageshadow="2"
     inkscape:zoom="0.7"
     inkscape:cx="72.892594"
     inkscape:cy="292.05522"
     inkscape:document-units="mm"
     inkscape:current-layer="layer1"
     showgrid="false"
     inkscape:window-width="1920"
     inkscape:window-height="1005"
     inkscape:window-x="-9"
     inkscape:window-y="-9"
     inkscape:window-maximized="1"
     fit-margin-top="0"
     fit-margin-left="0"
     fit-margin-right="0"
     fit-margin-bottom="0" />
  <metadata
     id="metadata5">
    <rdf:RDF>
      <cc:Work
         rdf:about="">
        <dc:format>image/svg+xml</dc:format>
        <dc:type
           rdf:resource="http://purl.org/dc/dcmitype/StillImage" />
        <dc:title></dc:title>
      </cc:Work>
    </rdf:RDF>
  </metadata>
  <g
     inkscape:label="Layer 1"
     inkscape:groupmode="layer"
     id="layer1"
     transform="translate(-42.581331,-53.491964)">
    <path
       style="fill:none;stroke:#180000;stroke-width:0.26458332px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1"
       d="m 68.688267,56.991964 c 1.158168,0 2.316337,0 3.474505,0 0.267269,0 0.539729,-0.05242 0.801807,0 0.195344,0.03907 0.339198,0.2282 0.53454,0.267269 0.223256,0.04465 2.022565,0 2.138156,0 0.44545,0 0.896959,0.07323 1.336347,0 0.277895,-0.04632 0.523915,-0.220954 0.801809,-0.267269 0.615143,-0.102524 1.252075,0.07735 1.870888,0 0.36449,-0.04556 0.704588,-0.221708 1.069078,-0.267269 0.353607,-0.0442 0.712718,0 1.069078,0 0.534537,0 1.069078,0 1.603615,0 1.870888,0 3.741775,0 5.612662,0 1.425438,0 2.850875,0 4.276312,0 0.386522,0 1.588654,-0.05645 1.870888,0 0.195342,0.03907 0.345551,0.204271 0.534537,0.267269 0.08452,0.02817 0.187585,-0.03984 0.267269,0 0.112692,0.05635 0.15458,0.210923 0.267272,0.267269 0.07028,0.03514 0.929317,0 1.069078,0 0.712718,0 1.425437,0 2.138156,0 2.227246,0 4.454486,0 6.681736,0"
       id="path3724"
       inkscape:connector-curvature="0" />
    <path
       style="fill:none;stroke:#180000;stroke-width:7;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       d="M 68.688267,56.991964 H 154.55576"
       id="path3732"
       inkscape:connector-curvature="0" />
    <path
       style="fill:none;stroke:#180000;stroke-width:7;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       d="m 69.810732,53.601642 c 0.0115,4.041206 -1.939463,8.768677 -2.620515,13.007515 -3.539262,22.028239 0.149691,39.882773 14.594257,57.467463 l 16.570711,21.38155 c 11.572005,11.8762 -0.6759,15.32041 1.870885,23.25246 l 3.20723,5.87992 c 2.29235,3.92317 0.81887,7.78401 0.009,9.8714 -0.94285,2.43009 -5.168998,5.19451 -5.168998,5.19451 0,0 -4.737763,1.60105 -6.051864,5.51941 l -1.884319,3.46894"
       id="path4539"
       inkscape:connector-curvature="0"
       sodipodi:nodetypes="csccccsccc" />
    <path
       style="fill:none;stroke:#180000;stroke-width:7;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       d="M 42.677862,70.78125 H 64.667378"
       id="path3732-8"
       inkscape:connector-curvature="0" />
    <path
       style="fill:none;stroke:#180000;stroke-width:7;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       d="m 46.079648,67.568452 c 0.10277,3.313056 0.300033,9.269491 0.308308,11.282075 l 1.014966,6.475888 c 2.071426,10.478826 10.427978,22.734615 15.387389,28.125375 5.674079,5.84121 17.96301,13.27915 24.522188,17.23868"
       id="path4556"
       inkscape:connector-curvature="0"
       sodipodi:nodetypes="ccccc" />
    <path
       style="fill:none;stroke:#180000;stroke-width:7;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       d="M 79.941963,200.0491 72.382439,217.625 149.1116,217.24702 142.11905,200.0491 Z"
       id="path4594"
       inkscape:connector-curvature="0"
       sodipodi:nodetypes="ccccc" />
    <path
       style="fill:none;stroke:#180000;stroke-width:7;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       d="m 152.16174,53.747783 c -0.0115,4.041206 1.93946,8.768677 2.62051,13.007515 3.53926,22.028238 -0.14969,39.882772 -14.59425,57.467462 l -16.57072,21.38155 c -11.572,11.8762 0.6759,15.32041 -1.87088,23.25246 l -3.20723,5.87992 c -2.29235,3.92317 -0.81887,7.78401 -0.009,9.8714 0.94285,2.43009 5.16899,5.19451 5.16899,5.19451 0,0 4.73777,1.60105 6.05187,5.51941 l 1.88432,3.46894"
       id="path4539-0"
       inkscape:connector-curvature="0"
       sodipodi:nodetypes="csccccsccc" />
    <path
       style="fill:none;stroke:#180000;stroke-width:7;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       d="M 179.29461,70.927391 H 157.30509"
       id="path3732-8-5"
       inkscape:connector-curvature="0" />
    <path
       style="fill:none;stroke:#180000;stroke-width:7;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       d="m 175.89282,67.714593 c -0.10277,3.313056 -0.30003,9.269491 -0.30831,11.282075 l -1.01496,6.475889 c -2.07143,10.478825 -10.42798,22.734613 -15.38739,28.125373 -5.67408,5.84121 -17.96301,13.27915 -24.52219,17.23868"
       id="path4556-7"
       inkscape:connector-curvature="0"
       sodipodi:nodetypes="ccccc" />
  </g>
</svg>
