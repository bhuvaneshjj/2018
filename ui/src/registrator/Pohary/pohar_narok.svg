<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!-- Created with Inkscape (http://www.inkscape.org/) -->

<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   width="136.8098mm"
   height="167.65932mm"
   viewBox="0 0 136.8098 167.65932"
   version="1.1"
   id="svg8"
   inkscape:version="0.92.3 (2405546, 2018-03-11)"
   sodipodi:docname="pohar_narok.svg">
  <defs
     id="defs2" />
  <sodipodi:namedview
     id="base"
     pagecolor="#ffffff"
     bordercolor="#666666"
     borderopacity="1.0"
     inkscape:pageopacity="0.0"
     inkscape:pageshadow="2"
     inkscape:zoom="0.98994949"
     inkscape:cx="-93.2775"
     inkscape:cy="292.05522"
     inkscape:document-units="mm"
     inkscape:current-layer="layer1"
     showgrid="false"
     inkscape:window-width="1920"
     inkscape:window-height="1005"
     inkscape:window-x="-9"
     inkscape:window-y="-9"
     inkscape:window-maximized="1"
     fit-margin-top="0"
     fit-margin-left="0"
     fit-margin-right="0"
     fit-margin-bottom="0" />
  <metadata
     id="metadata5">
    <rdf:RDF>
      <cc:Work
         rdf:about="">
        <dc:format>image/svg+xml</dc:format>
        <dc:type
           rdf:resource="http://purl.org/dc/dcmitype/StillImage" />
        <dc:title />
      </cc:Work>
    </rdf:RDF>
  </metadata>
  <g
     inkscape:label="Layer 1"
     inkscape:groupmode="layer"
     id="layer1"
     transform="translate(-42.581331,-53.491964)">
    <path
       id="path4556-7"
       style="fill:none;stroke:#180000;stroke-width:7;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       d="m 175.89282,67.714593 c -0.10277,3.313056 -0.30003,9.269491 -0.30831,11.282075 l -1.01496,6.475889 c -2.07143,10.478825 -10.42798,22.734613 -15.38739,28.125373 -5.67408,5.84121 -17.96301,13.27915 -24.52219,17.23868 m 44.63464,-59.909219 h -21.98952 m -5.14335,-17.179608 c -0.0115,4.041206 1.93946,8.768677 2.62051,13.007515 3.53926,22.028238 -0.14969,39.882772 -14.59425,57.467462 l -16.57072,21.38155 c -11.572,11.8762 0.6759,15.32041 -1.87088,23.25246 l -3.20723,5.87992 c -2.29235,3.92317 -0.81887,7.78401 -0.009,9.8714 0.94285,2.43009 5.16899,5.19451 5.16899,5.19451 0,0 4.73777,1.60105 6.05187,5.51941 l 1.88432,3.46894 M 79.941963,200.0491 72.382439,217.625 149.1116,217.24702 142.11905,200.0491 Z M 46.079648,67.568452 c 0.10277,3.313056 0.300033,9.269491 0.308308,11.282075 l 1.014966,6.475888 c 2.071426,10.478826 10.427978,22.734615 15.387389,28.125375 5.674079,5.84121 17.96301,13.27915 24.522188,17.23868 M 42.677862,70.78125 h 21.989516 m 5.143354,-17.179608 c 0.0115,4.041206 -1.939463,8.768677 -2.620515,13.007515 -3.539262,22.028239 0.149691,39.882773 14.594257,57.467463 l 16.570711,21.38155 c 11.572005,11.8762 -0.6759,15.32041 1.870885,23.25246 l 3.20723,5.87992 c 2.29235,3.92317 0.81887,7.78401 0.009,9.8714 -0.94285,2.43009 -5.168998,5.19451 -5.168998,5.19451 0,0 -4.737763,1.60105 -6.051864,5.51941 l -1.884319,3.46894 M 68.688267,56.991964 h 85.867493 m -85.867493,0 c 1.158168,0 2.316337,0 3.474505,0 0.267269,0 0.539729,-0.05242 0.801807,0 0.195344,0.03907 0.339198,0.2282 0.53454,0.267269 0.223256,0.04465 2.022565,0 2.138156,0 0.44545,0 0.896959,0.07323 1.336347,0 0.277895,-0.04632 0.523915,-0.220954 0.801809,-0.267269 0.615143,-0.102524 1.252075,0.07735 1.870888,0 0.36449,-0.04556 0.704588,-0.221708 1.069078,-0.267269 0.353607,-0.0442 0.712718,0 1.069078,0 0.534537,0 1.069078,0 1.603615,0 1.870888,0 3.741775,0 5.612662,0 1.425438,0 2.850875,0 4.276312,0 0.386522,0 1.588654,-0.05645 1.870888,0 0.195342,0.03907 0.345551,0.204271 0.534537,0.267269 0.08452,0.02817 0.187585,-0.03984 0.267269,0 0.112692,0.05635 0.15458,0.210923 0.267272,0.267269 0.07028,0.03514 0.929317,0 1.069078,0 0.712718,0 1.425437,0 2.138156,0 2.227246,0 4.454486,0 6.681736,0" />
    <path
       style="fill:#ccf0ce;stroke:#470000;stroke-width:0;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1;fill-opacity:1"
       d="m 201.26153,538.24398 c 1.72868,-3.18734 6.58315,-7.05696 12.74937,-10.16286 3.33351,-1.67908 9.11023,-5.58526 12.83716,-8.68044 13.95355,-11.58823 19.90426,-23.64528 20.73803,-42.01839 0.58834,-12.96467 -0.64145,-17.14773 -10.28345,-34.97846 -7.91982,-14.64597 -8.17866,-16.88827 -3.15872,-27.36314 8.22198,-17.15637 10.72896,-28.35902 8.76762,-39.17873 -1.38948,-7.66509 -7.75321,-20.57016 -13.69845,-27.77919 -2.52,-3.05572 -17.16518,-21.69303 -32.54483,-41.41626 -15.37964,-19.72323 -32.93205,-42.21648 -39.00536,-49.98498 -25.11219,-32.12162 -39.00735,-58.28371 -47.59234,-89.60783 -6.19397,-22.60002 -7.17787,-31.24393 -7.16451,-62.94277 0.014,-33.280097 1.66898,-47.405412 8.58052,-73.236501 l 0.94611,-3.535971 145.945,-0.296833 145.94501,-0.296832 1.00352,3.832803 c 5.78788,22.106226 7.77335,34.978569 9.17601,59.490826 3.04608,53.231518 -8.96547,99.336558 -37.0811,142.331878 -7.08753,10.83848 -66.53541,88.02496 -88.56737,114.9949 -7.83543,9.59158 -12.30332,18.35831 -14.34393,28.14512 -2.39765,11.4992 -0.84225,19.34554 7.44755,37.56958 2.50504,5.50701 4.58362,11.62947 4.62978,13.63706 0.0623,2.71135 -1.75577,7.00219 -7.10802,16.77522 -8.46849,15.46318 -10.41491,21.40453 -10.41491,31.79092 0,8.07107 1.38258,14.73833 5.14121,24.79252 3.1952,8.54706 17.14111,22.35257 29.09857,28.80563 4.66448,2.51728 9.60934,6.04019 10.9886,7.82869 l 2.50771,3.25181 h -58.24877 c -55.16683,0 -58.19805,-0.0935 -57.29001,-1.76777 z"
       id="path857"
       inkscape:connector-curvature="0"
       transform="matrix(0.26458333,0,0,0.26458333,42.581331,53.491964)" />
    <path
       style="fill:#ccf0ce;stroke:#470000;stroke-width:0;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1;fill-opacity:1"
       d="M 100.36874,229.6755 C 90.71866,222.03044 82.626759,213.49392 73.565767,201.39982 57.425966,179.85732 42.821289,153.34929 36.215078,133.60691 32.24206,121.73372 27.668203,96.418395 27.649944,86.200718 l -0.01218,-6.818529 H 51.319682 75.0016 l -0.02799,22.980971 c -0.03083,25.3079 1.519357,41.50399 5.825287,60.86169 4.628962,20.80992 13.132941,43.69602 23.174353,62.36732 2.4622,4.57829 4.47672,8.55576 4.47672,8.83884 0,1.25028 -1.94531,0.10554 -8.08122,-4.75551 z"
       id="path859"
       inkscape:connector-curvature="0"
       transform="matrix(0.26458333,0,0,0.26458333,42.581331,53.491964)" />
    <path
       style="fill:#ccf0ce;stroke:#470000;stroke-width:0;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1;fill-opacity:1"
       d="m 407.45512,236.61241 c 0,-0.26351 1.86555,-3.71486 4.14567,-7.66969 9.6925,-16.81142 19.79172,-43.46862 24.51484,-64.70772 4.6542,-20.92917 6.02698,-34.82121 6.0142,-60.86169 l -0.0118,-23.991121 h 23.57971 23.57971 v 7.478812 c 0,10.340082 -4.5651,36.412069 -8.21718,46.929599 -10.27316,29.58537 -34.86131,68.84707 -55.15442,88.06907 -8.16904,7.73786 -18.45075,15.95885 -18.45075,14.75274 z"
       id="path861"
       inkscape:connector-curvature="0"
       transform="matrix(0.26458333,0,0,0.26458333,42.581331,53.491964)" />
    <path
       style="fill:#ccf0ce;stroke:#470000;stroke-width:0;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1;fill-opacity:1"
       d="m 133.88326,605.77846 c 0.17907,-0.46015 3.94836,-9.30922 8.3762,-19.66461 l 8.0506,-18.82798 H 258.5603 366.81054 l 7.70499,18.84657 c 4.23774,10.36562 7.58702,18.93972 7.44284,19.05355 -0.86587,0.68364 -248.34061,1.27468 -248.07511,0.59247 z"
       id="path863"
       inkscape:connector-curvature="0"
       transform="matrix(0.26458333,0,0,0.26458333,42.581331,53.491964)" />
  </g>
</svg>
