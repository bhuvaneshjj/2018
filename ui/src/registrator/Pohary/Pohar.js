import React from 'react';
import PropTypes from 'prop-types';
import './Pohar.css';
import poharNarok from './pohar_narok.svg';
import poharNeprevzato from './pohar_neprevzato.svg';
import poharPredano from './pohar_predano.svg';

const styleToImg = {
  nárok: poharNarok,
  nepřevzato: poharNeprevzato,
  předáno: poharPredano
};

const Pohar = ({ poharStyle, sizePercentage }) => (
  <img
    src={styleToImg[poharStyle]}
    alt={`pohár - ${poharStyle}`}
    className="Pohar"
    title={`pohár - ${poharStyle}`}
    width={(sizePercentage * 517) / 100}
    height={(sizePercentage * 643) / 100}
  />
);

Pohar.propTypes = {
  poharStyle: PropTypes.oneOf(['nárok', 'nepřevzato', 'předáno']).isRequired,
  sizePercentage: PropTypes.number.isRequired
};

Pohar.defaultProps = {
  sizePercentage: 100
};

export default Pohar;
