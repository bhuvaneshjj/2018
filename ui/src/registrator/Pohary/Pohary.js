import React from 'react';
import PropTypes from 'prop-types';
import Pohar from './Pohar';

const Pohary = ({ count, poharStyle }) => (
  <div>
    {Array(count)
      .fill(1)
      .map((val, index) => (
        <Pohar key={index} poharStyle={poharStyle} sizePercentage={4} />
      ))}
  </div>
);

Pohary.propTypes = {
  count: PropTypes.number.isRequired,
  poharStyle: PropTypes.oneOf(['nárok', 'nepřevzato', 'předáno']).isRequired
};

export default Pohary;
