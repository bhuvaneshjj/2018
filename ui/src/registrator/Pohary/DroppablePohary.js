import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { DropTarget } from 'react-dnd';
import { DragTypes } from '../../constants';
import { TYPY_POHARU } from './PoharyActions';
import Pohary from './Pohary';

const collect = (connect, monitor) => ({
  connectDropTarget: connect.dropTarget(),
  isDropOver: monitor.isOver() && monitor.canDrop()
});

class DroppablePohary extends PureComponent {
  constructor(props) {
    super(props);
    const { type, canDrop, onDrop } = this.props;

    this.dropTarget = {
      drop: (dropProps, monitor) =>
        onDrop({ source: monitor.getItem(), destination: { ...dropProps, type } }),
      canDrop: (dropProps, monitor) =>
        canDrop({ source: monitor.getItem(), destination: { ...dropProps, type } })
    };
  }

  render = () => {
    const { canDrop, onDrop, ...rest } = this.props;
    const Droppable = DropTarget(DragTypes.POHAR, this.dropTarget, collect)(Pohary);
    return <Droppable {...rest} />;
  };
}

DroppablePohary.propTypes = {
  type: PropTypes.oneOf(TYPY_POHARU).isRequired,
  canDrop: PropTypes.func.isRequired,
  onDrop: PropTypes.func.isRequired
};

export default DroppablePohary;
