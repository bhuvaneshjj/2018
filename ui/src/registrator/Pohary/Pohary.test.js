import React from 'react';
import renderer from 'react-test-renderer';
import Pohary from './Pohary';

it('dva nepřevzaté poháry', () => {
  const component = renderer.create(<Pohary count={2} poharStyle="nepřevzato" />);
  expect(component.toJSON()).toMatchSnapshot();
});
