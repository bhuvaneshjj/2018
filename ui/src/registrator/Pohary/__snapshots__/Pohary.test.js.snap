// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dva nepřevzaté poháry 1`] = `
<div>
  <img
    alt="pohár - nepřevzato"
    className="Pohar"
    height={25.72}
    src="pohar_neprevzato.svg"
    title="pohár - nepřevzato"
    width={20.68}
  />
  <img
    alt="pohár - nepřevzato"
    className="Pohar"
    height={25.72}
    src="pohar_neprevzato.svg"
    title="pohár - nepřevzato"
    width={20.68}
  />
</div>
`;
